gdnative-sys/build.rs

```diff
@@ -91,29 +91,91 @@ mod header_binding {
             .to_string_lossy(),
         );
 
+        let mut android_ndk_root: Option<PathBuf> = None;
+
+        if Path::join(&android_sdk_root, "ndk").exists() {
+            // New NDK
+            let available_ndk_versions: Vec<_> =
+                std::fs::read_dir(Path::join(&android_sdk_root, "ndk"))
+                    .unwrap()
+                    .into_iter()
+                    .map(|dir| dir.unwrap().path())
+                    .collect();
+
+            if !available_ndk_versions.is_empty() {
+                let ndk_version = std::env::var("ANDROID_NDK_VERSION");
+
+                if let Ok(ndk_version) = ndk_version {
+                    if available_ndk_versions
+                        .iter()
+                        .map(|p| p.file_name())
+                        .any(|p| {
+                            p.is_some() && p.unwrap().to_string_lossy().eq(ndk_version.as_str())
+                        })
+                    {
+                        // Asked version is available
+                        android_ndk_root = Some(Path::join(
+                            &Path::join(&android_sdk_root, "ndk"),
+                            ndk_version,
+                        ))
+                    } else {
+                        panic!(
+                            "no available android ndk versions matches {}. Available versions : {:?}",
+                            ndk_version, available_ndk_versions
+                        )
+                    }
+                } else {
+                    // No NDK version chosen, chose the most recent one and issue a warning
+                    println!("cargo:warning=Multiple android ndk versions have been detected.");
+                    println!("cargo:warning=You should chose one using ANDROID_NDK_VERSION environment variable to have reproducible builds.");
+                    println!(
+                        "cargo:warning=Available versions : {:?}",
+                        available_ndk_versions
+                    );
+
+                    let ndk_version = available_ndk_versions
+                        .iter()
+                        .filter_map(|p| p.file_name())
+                        .filter_map(|v| semver::Version::parse(v.to_string_lossy().as_ref()).ok())
+                        .max()
+                        .unwrap();
+
+                    println!(
+                        "cargo:warning=Automatically chosen version : {} (latest)",
+                        ndk_version
+                    );
+
+                    android_ndk_root = Some(Path::join(
+                        &Path::join(&android_sdk_root, "ndk"),
+                        ndk_version.to_string(),
+                    ));
+                }
+            }
+        }
+
+        if android_ndk_root.is_none() && Path::join(&android_sdk_root, "ndk-bundle").exists() {
```

Comment:
I'd also extract `Path::join(&android_sdk_root, "ndk-bundle")` to a variable.

It might need to be unnecessarily constructed if the first condition is false, but the code is more readable and the path itself is not constructed twice.